## __BME 580.707 Algorithms for Analysis of Genomic Sequence Data.__
 
__Course Description__: This course is aimed at introducing students to the concepts of computational bioinformatics and genomics. Emphasis will be placed on algorithms and data structures for genomic data. The course will cover how genomic and transcriptomic data are produced, data file formats, and basic tools for processing the data. It will also include algorithms used in the alignment and assembly of genomic sequence data, with hands-on computational examples, followed by topics in genome sequence analysis and annotation.

__Professor__: Alexey Zimin,  https://ccb.jhu.edu/people/alekseyz/
__Times__: MWF, 8:00am â€“ 8:50am 
Location: TBA
Office hours and location: Wyman Park building, second floor, S237, Wednesdays 4pm-5pm.
Textbook: None, all lecture slides will be made available online, all slides will be posted at the end of each week at 
Attendance policy: attendance is mandatory, no video/audio lecture recordings will be made available online. Lecture slides will be available online.
GRADING POLICY: Four laboratory assignments are worth 15% each. The in-class presentation is worth 10%. The final project is worth 30%.

SYLLABUS
Note: this is a new course and this schedule may change during the semester; refresh your browser for updates. Changes will be posted here or announced in class. 
Week 1, August 25,27,29: Introduction to the course. Course logistics.  Biology and genomics background, genome sequencing technology. Genomic data and storage formats. NCBI Genbank, RefSeq and SRA.  
Week 2, September 3 and 5: The concept of coverage, probabilistic estimations. Genomic sequence alignment, Smith-Waterman alignment algorithm.
Week 3, September 8,10,12: Data Structure: Suffix Array. Alignment of whole genomes. Concept of k-mers and minimizers. Seeding an alignment: method of minimizers.  Lab 1 assigned: compute minimizers for genomic sequences.
Week 4, September 15,16,17: Lab 1 due on September 17. Q and A on Lab 1. Alignment of a short query against a genome. Data structure: Suffix array. Burrows-Wheeler transform. Overview of alignment tools and alignment file formats: blast, nucmer, BWA, BowTie, minimap.
Week 5, September 22,24,26: Genome assembly approaches: Overlap-Layout-Consensus and DeBruijn graph. Data Structure: k-mer graph. Concept of unitigs. Lab 2 assigned: build unitigs from reads.
Week 6, September 29, October 1, 3: Lab 2 due on October 3. Q and A on Lab 2. Assembly of short reads and hybrid assembly of short and long reads. Types of genomic data and assembly programs. 
Week 7, October 6,8,10: Evaluation of genome assemblies.  K-mer and alignment-based approaches.  Polishing genome assemblies. Lab 3 assigned: genome assembly.
Week 8, October 13,15,17: Lab 3 due on October 17. Q and A on Lab 3. Graph structures: Directed Acyclic graph and its applications in genome assembly.
Week 9, October 20,22,24: DNA expression, transcription, splicing and translation. Genome Annotation: making sense of the genomic sequence. Annotation file formats.
Week 10, October 27,29,31: Transcriptome data types, transcript assembly tools, evaluation of transcript assemblies. Lab 4 assigned: assembly of RNA-seq data.
Week 11, November 3,5,7: Lab 4 due on November 7. Q and A on Lab 4. Introduction to Markov chain models and Convolution Neural Networks for genome annotation.
Week 12, November 10,12,14: Genome annotation tools: ab initio and mapping-based. Introduction to Hidden Markov Models (HMMs) and their application to genome annotation. Annotation liftover.
Week 13, November 17,19,21: Quantification of genome expression: approaches and tools.
Week 14, December 1,3,5: Topics in genome sequencing, assembly and annotation, TBA. Final project assigned: assemble and annotate a genome; quantify expresstion.
Week 15, December 8,10,12: Student presentations of selected papers (10-15 minutes per 
presentation, 3-4 per class).
Week 16, December 15: Q and A on the final project. Final Project Due December 17, 11:59pm.

